catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java

```diff
@@ -183,12 +183,16 @@ public boolean sourceStatus(String servicePID) {
                 if (configs != null) {
                     for (Configuration config : configs) {
                         ConfigurationDetails source = new ConfigurationDetailsImpl();
-
-                        boolean disabled = config.getPid()
-                                .endsWith(ConfigurationStatus.DISABLED_EXTENSION);
-                        source.setId(config.getPid());
+                        String pid = config.getPid();
+                        String fPid = config.getFactoryPid();
+                        boolean disabled = pid.endsWith(ConfigurationStatus.DISABLED_EXTENSION);
+                        if (fPid != null) {
+                            disabled = disabled
+                                    || fPid.endsWith(ConfigurationStatus.DISABLED_EXTENSION);
+                        }
```

Comment:
`boolean disabled = pid.endsWith(ConfigurationStatus.DISABLED_EXTENSION) 
                || ( fPid != null && fPid.endsWith(ConfigurationStatus.DISABLED_EXTENSION));`
Would this be simpler?